project_task_material_stock/models/project_task.py

```diff
@@ -177,10 +183,29 @@ def _prepare_stock_move(self):
 
     @api.multi
     def create_stock_move(self):
-        for line in self:
-            move_id = self.env['stock.move'].create(
-                line._prepare_stock_move())
-            line.stock_move_id = move_id.id
+        pick_type = self.env.ref('stock.picking_type_out')
+
+        if self:
+            task = self[0].task_id
+            location_id = task.location_source_id.id or task.project_id.location_source_id.id or self.env.ref(
+                'stock.stock_location_stock').id
+            location_dest_id = task.location_dest_id.id or task.project_id.location_dest_id.id or self.env.ref(
+                'stock.stock_location_customers').id
+
+            picking_id = task.picking_id or self.env['stock.picking'].create({
```

Comment:
picking_id is not necessary if you follow my instructions bellow.

Reply:
if you need add a new material in task? picking id it's necessary?
I need you  help in this point. We should be able to work with partial materials in tasks